tests/test_sample_rayservice_yamls.py

```diff
@@ -1,55 +1,338 @@
 ''' Test sample RayService YAML files to catch invalid and outdated ones. '''
-import unittest
-import os
+from copy import deepcopy
+from kubernetes import client
 import logging
+import os
+import pytest
+import sys
+from tempfile import NamedTemporaryFile
+import time
+from typing import Any, Dict, List, Optional
 import yaml
 
 from framework.prototype import (
     RuleSet,
-    GeneralTestCase,
-    RayServiceAddCREvent,
+    CREvent,
     EasyJobRule,
-    CurlServiceRule
+    CurlServiceRule,
+    AutoscaleRule,
+    get_expected_head_pods,
+    get_expected_worker_pods,
+    show_cluster_info,
+    check_pod_running,
 )
 
 from framework.utils import (
-    CONST
+    start_curl_pod,
+    logger,
+    shell_subprocess_run,
+    CONST,
+    K8S_CLUSTER_MANAGER,
+    OperatorManager
 )
 
 logger = logging.getLogger(__name__)
 
-if __name__ == '__main__':
-    NAMESPACE = 'default'
-    SAMPLE_PATH = CONST.REPO_ROOT.joinpath("ray-operator/config/samples/")
-    YAMLs = ['ray_v1alpha1_rayservice.yaml']
-
-    sample_yaml_files = []
-    for filename in YAMLs:
-        filepath = SAMPLE_PATH.joinpath(filename)
-        with open(filepath, encoding="utf-8") as cr_yaml:
-            for k8s_object in yaml.safe_load_all(cr_yaml):
-                sample_yaml_files.append(
-                    {'path': filepath, 'name': filename, 'cr': k8s_object}
-                )
-
-    rs = RuleSet([EasyJobRule(), CurlServiceRule()])
-    image_dict = {
-        CONST.RAY_IMAGE_KEY: os.getenv('RAY_IMAGE', default='rayproject/ray:2.5.0'),
-        CONST.OPERATOR_IMAGE_KEY: os.getenv('OPERATOR_IMAGE', default='kuberay/operator:nightly'),
-    }
-    logger.info(image_dict)
-
-    # Build a test plan
-    logger.info("Build a test plan ...")
-    test_cases = unittest.TestSuite()
-    for index, new_cr in enumerate(sample_yaml_files):
-        logger.info('[TEST %d]: %s', index, new_cr['name'])
-        addEvent = RayServiceAddCREvent(new_cr['cr'], [rs], 90, NAMESPACE, new_cr['path'])
-        test_cases.addTest(GeneralTestCase('runtest', image_dict, addEvent))
-
-    # Execute all tests
-    runner = unittest.TextTestRunner()
-    test_result = runner.run(test_cases)
-
-    # Without this line, the exit code will always be 0.
-    assert test_result.wasSuccessful()
+NAMESPACE = 'default'
+DEFAULT_IMAGE_DICT = {
+    CONST.RAY_IMAGE_KEY: os.getenv('RAY_IMAGE', default='rayproject/ray:2.5.0'),
```

Comment:
Should we use the `nightly` instead of 2.5.0 here?

Reply:
Hmm not sure what the convention is, but the Kuberay and Anyscale tests I've seen are run against the latest version of Ray.

Reply:
Got it– in that case, let's leave the version as is.